### ISSUE TITLE
Bug: no-restricted-imports group matches what is written in file instead of relative to root?

### ISSUE THREAD

**Issue by AnderssonPeter** [eslint/eslint#15860]
### Environment

Node version: 14.17.6
npm version: 6.14.15
Local ESLint version: 8.10.0
Global ESLint version: 8.10.0
Operating System: Windows 10 20H2

### What parser are you using?

Default (Espree) (I'm not 100% sure I'm using angular 13)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "files": [
    "*.ts"
  ],
  "rules": {
    "no-restricted-imports": ["error", {
      "patterns": [{
        "group": ["*/ui/*"],
        "message": "ui services should not be used!"
      }]
    }]
  }
},
```
</details>

Lets say my file is in the ui folder then I can have 2 different types of imports, one that is relative from my current file and one that specifics the location from the root.

The file below is inside the ui folder.
```ts
import { DialogService1 } from 'ui/dialog.service'; //This will trigger the linting
import { DialogService2 } from './dialog.service'; //This will not trigger the linting 
```

### What did you expect to happen?

I expected the linting to trigger on both lines

### What actually happened?

It only triggers on the line that's uses the full path and does not trigger on the one with a relative path.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by AnderssonPeter** [comment]
Changing to `"group": ["**/ui/**"],` resolved my issue!
